rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -1126,14 +1126,29 @@ stream {
             {{ if not (isLocationInLocationList $location $all.Cfg.NoTLSRedirectLocations) }}
             # enforce ssl on server side
             if ($redirect_to_https) {
+                set $redirect_port {{ printf ":%v" $all.ListenPorts.HTTPS }};
+                set_by_lua_block $redirect_host {
+                    local ngx_re = require "ngx.re"
+
+                    local host_port, err = ngx_re.split(ngx.var.best_http_host, ":")
+                    if err then
+                        ngx.log(ngx.ERR, "could not parse variable: ", err)
+                        return ngx.var.best_http_host;
+                    end
+
+                    if host_port[0] ~= nil then
+                        return host_port[0]
+                    else
+                        return host_port[1]
+                    end;
```

Comment:
Why's the else condition here? Why not only `return host_port[0]`?

Reply:
fixed

Reply:
that's because in Lua indexes start from 1, so you need to `return host_port[1]`